sample-pong/src/commonMain/kotlin/PlayScene.kt

```diff
@@ -0,0 +1,213 @@
+                            "-- To move the Left Paddle, use the keys [W] and [S].\n\n" +
+                            "-- To move the Right Paddle, use the keys [UP] and [DOWN]\n\n" +
+                            "-- To go back to Main Menu, use [ESC]\n\n" +
+                            "-- To Start the game, use [SPACE]"
+                // show a different text if the game is in Scored state
+                } else if (playState == GameStates.Scored) {
+                    text = "Press [SPACE] for the next round!"
+                }
+            }
+        }
+
+        // A simple flavour text informing that a goal was scored
+        scoredYellText = text("SCORED!!!\n\n") {
+            position(views.actualVirtualWidth / 2 - 100, views.actualVirtualHeight / 2 - 20)
+            addUpdater {
+                // this is only visible when the game is in Scored state
+                visible = (playState == GameStates.Scored)
+            }
+        }
+
+        // text to show the score of the player on the Left side
+        scoreLeftText = text("0") {
+//            textSize = 24.0
+            position(views.actualVirtualWidth / 4, views.actualVirtualHeight / 2)
+            addUpdater {
+                text = scorePlayerLeft.toString()
+            }
+        }
+        // text to show the score of the player on the Right side
+        scoreRightText = text("0") {
+            position(views.actualVirtualWidth * 3 / 4, views.actualVirtualHeight / 2)
+            addUpdater {
+                text = scorePlayerRight.toString()
+            }
+        }
+
+        // the left paddle
+        paddleLeft = solidRect(paddleWidth, paddleHeight, Colors.RED) {
+            position(paddleDistanceFromWall, paddlePosYAtStart)
+            addUpdater {
+                // move the paddle up or down as long as it doesn't leaves the bounds of the game window
+                val keys = views.input.keys
+                if (keys[Key.W] && y > 0) {
+                    y -= paddleMoveSpeed
+                }
+                if (keys[Key.S] && y < views.actualVirtualHeight - paddleHeight) {
+                    y += paddleMoveSpeed
+                }
+            }
+        }
+
+        // the right paddle
+        paddleRight = solidRect(paddleWidth, paddleHeight, Colors.BLUE) {
+            position(views.actualVirtualWidth - paddleDistanceFromWall - paddleWidth, paddlePosYAtStart)
+            addUpdater {
+                // move the paddle up or down as long as it doesn't leaves the bounds of the game window
+                val keys = views.input.keys
+                if (keys[Key.UP] && y > 0) {
+                    y -= paddleMoveSpeed
+                }
+                if (keys[Key.DOWN] && y < views.actualVirtualHeight - paddleHeight) {
+                    y += paddleMoveSpeed
+                }
+            }
+        }
+
+        ball = circle(ballRadius, Colors.WHITE) {
+            position(ballPosXAtStart, ballPosYAtStart)

```

Comment:
You have quite a big nesting below with many lines of code. I suggest you extract it into a function or move it outside `circle () { ... }` at least. This way it will be much easier to read and understand.